doc/manual/src/development/debugging.md

```diff
@@ -0,0 +1,61 @@
+# Debugging Nix
+
+This section provides instructions on how to build and debug Nix with debug
+symbols enabled. It assumes you are using Nix with the [`flakes`] and
+[`nix-command`] experimental features enabled.
+
+[`flakes`]: @docroot@/development/experimental-features.md#xp-feature-flakes
+[`nix-command`]: @docroot@/development/experimental-features.md#xp-nix-command
+
+## Building Nix with Debug Symbols
+
+First, ensure you have set up the development environment as described in the
+[building documentation](./building.md).
+
+In the development shell, set the `mesonBuildType` environment variable to
+`debug` before configuring the build:
+
+```console
+[nix-shell]$ export mesonBuildType=debug
+```
+
+Then, configure and build Nix:
+
+```console
+[nix-shell]$ mesonConfigurePhase
+[nix-shell]$ ninjaBuildPhase
+```
+
+This will build Nix with debug symbols, which are essential for effective
+debugging.
+
+## Debugging the Nix Binary
+
+### Installing a Debugger
+
+Install your preferred debugger within the development shell. For example, to
+install `lldb`:
+
+```console
+[nix-shell]$ nix shell nixpkgs#lldb
+```
+
+### Launching the Debugger
+
+To debug the Nix binary you just built:
+
+```console
+[nix-shell]$ lldb -- ./subprojects/nix/nix
```

Comment:
FWIW, I use `gdb`, and if you're in a `nix develop` shell, you can just do
```
gdb --args nix ...
```
i.e. no need to give a full path to the `nix` binary.

Reply:
So meson actually installs it to /home/joerg/git/nix/out/bin which is not in PATH. But in that case we would also need to run `ninjaInstallPhase`. I think it makes sense that both build system choose different locations for our own sanity while we still have two and not put both in PATH. I actually prefer being explicit about the executable location, because it's less error prone for users following the tutorial. Let's say one person has exited the devshell to get a debugger and no longer has the PATH from it set, in that case they might run the debugger against the global installed nix, which is not good.

Reply:
Yes, I can confirm @Mic92 comment.